metricflow/test/model/validations/test_measures.py

```diff
@@ -332,3 +336,90 @@ def test_count_measure_with_distinct_expr(simple_model__pre_transforms: UserConf
     assert (
         actual_error.find(expected_error_substring) != -1
     ), f"Expected error {expected_error_substring} not found in error string! Instead got {actual_error}"
+
+
+def test_percentile_measure_missing_agg_params(simple_model__pre_transforms: UserConfiguredModel) -> None:
+    """Tests that all measures with COUNT agg should have expr provided."""
+    model = copy.deepcopy(simple_model__pre_transforms)
+    data_source = find_data_source_with(model, lambda ds: ds.name == "bookings_source")[0]
+    invalid_measure_1 = Measure(
+        name="bad_measure_1",
+        agg=AggregationType.PERCENTILE,
+        agg_time_dimension="ds",
+    )
+    invalid_measure_2 = Measure(
+        name="bad_measure_2",
+        agg=AggregationType.PERCENTILE,
+        agg_time_dimension="ds",
+        agg_params=MeasureAggregationParameters(),
+    )
+
+    invalid_measure_3 = Measure(
+        name="bad_measure_3",
+        agg=AggregationType.SUM,
+        agg_time_dimension="ds",
+        agg_params=MeasureAggregationParameters(percentile=0.3),
+    )
+    data_source.measures.extend([invalid_measure_1, invalid_measure_2, invalid_measure_3])  # type: ignore
```

Comment:
I'd like to move away from this pattern and on to one where we make a simple data source that has the measures we need, just for validation. See https://github.com/transform-data/metricflow/blob/main/metricflow/test/model/validations/test_validity_param_definitions.py#L11-L39 for an example.